tests/test_format_alerts.py

```diff
@@ -0,0 +1,213 @@
+            # temp_file now contains the reformatted alert
+
+    """
+    survey, version, __ = path.stem.split('_')
+
+    with open(path, 'rb') as f:
+        temp_file.write(f.read())
+    temp_file.seek(0)
+    consume.GCSKafkaConsumer.fix_schema(temp_file, survey, version)
+
+    temp_file.seek(0)
+
+
+class AlertFormattingDataUnchanged(TestCase):
+    """Test data we retrieve from file before / after formatting with
+    ``consume.GCSKafkaConsumer.fix_schema()`` is the same.
+    """
+
+    def test_data_unchanged_ztf_3_3(self):
+        """ ZTF version 3.3 formatting, test that data is unchanged. """
+
+        survey, version = 'ztf', '3.3'
+        self.data_unchanged(test_alert_path[f'{survey}_{version}'])
+
+    def data_unchanged(self, path: Path):
+        """ Tests that the data in the path file is unchanged when reformatting
+        the schema header with `consume.GCSKafkaConsumer.fix_schema()`.
+        """
+        survey, version, __ = path.stem.split('_')
+
+        # Get the original data
+        __, original_data = _load_Avro(str(path))
+
+        # Correct the schema and get the data again
+        with consume.TempAlertFile(max_size=max_alert_size, mode='w+b') as temp_file:
+            run_consume_fix_schema(path, temp_file)
+            __, corrected_data = _load_Avro(temp_file)
+
+        # test data unchanged
+        msg = f'Data was changed while correcting the schema header for {survey} version {version}'
+        self.assertEqual(original_data, corrected_data, msg)
+
+
+class AlertFormattedForBigQuery(TestCase):
+    """ Test that the alerts are formatted properly for import into BigQuery.
+    This test clears the BQ test table before uploading an alert and therefore
+    does _NOT_ test whether different survey/version combinations are compatible
+    for upload to the same BQ table.
+
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.client = bigquery.Client()
+        cls.dataset_id = dataset_id
+        cls.table_id = 'temp_table' # created automatically if it doesn't yet exist
+
+        cls.dataset_ref = cls.client.dataset(cls.dataset_id)
+        cls.table_ref = cls.dataset_ref.table(cls.table_id)
+        cls.job_config = bigquery.LoadJobConfig()
+        cls.job_config.source_format = bigquery.SourceFormat.AVRO
+        cls.job_config.write_disposition = bigquery.WriteDisposition.WRITE_TRUNCATE
+        cls.job_config.autodetect = True # enable schema autodetection
+
+
+    def test_BQupload_ztf_3_3(self):
+        """ ZTF version 3.3 upload to BigQuery test. """
+
+        survey, version = 'ztf', '3.3'
+        path = test_alert_path[f'{survey}_{version}']
+
+        # Correct the schema and upload to BQ
+        with consume.TempAlertFile(max_size=max_alert_size, mode='r+b') as temp_file:
+            run_consume_fix_schema(path, temp_file)
+            self.upload_to_BigQuery(temp_file)
+
+
+    def upload_to_BigQuery(self, temp_file):
```

Comment:
may want to rename this to start with the word `assert` so it's clear this is design to potentially fail, error, or pass.